src/sensor/sensor.h

```diff
@@ -23,14 +23,22 @@ class Sensor : public QObject
     Q_INVOKABLE void connectLink(const QString& connString);
     Q_INVOKABLE void connectLinkLog(const QString& connString);
 
+    Q_PROPERTY(bool autodetect READ autoDetect WRITE setAutoDetect NOTIFY autoDetectUpdate)
+    void setAutoDetect(bool autodetect) { _autodetect = autodetect; emit autoDetectUpdate(autodetect); };
```

Comment:
Sorry, I think the return should go here so we do not set the field.

Is connecting/signaling inside of an object in order to accomplish logic like this a common pattern in Qt? Personally, I would rather have the logic in the lambda function put right here in this method.